### PR TITLE
add cypress-pipe, fix flaky test

### PR DESCRIPTION
**Pull request by aitchiss**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [ ] Bug Fix
- [X] Optimization
- [ ] Documentation Update

## Description

The `adjustPostTags` spec had been occasionally flaking in CI, due to a race condition between the Cypress click event, and the click handler being added asynchronously in `initializeActionsPanelToggle.js`. 

This kind of race condition has tripped us up a few times, and we've solved it ad hoc around the code by adding data attributes when click handlers are attached (so we can then await the data attribute being present in the DOM). This workaround however is far from ideal, leaving random data attributes around our production HTML.

Clicks aren't retriable in Cypress, because normally if X action doesn't occur on click, it should represent a test failure. However they point towards a package called `cypress-pipe` to account for these kind of situations where listeners are attached async. [See their blog post on the same](https://www.cypress.io/blog/2019/01/22/when-can-the-test-click/).

I've added that package to our dev dependencies, which lets us retry an action until a particular condition is met. The test step retries and times out/fails according the assertion's standard timeout. 

If we're in agreement to add this package in, I'll add to the developer documentation / write up a post explaining how and when we would use this. I'll also look at places where we're using the old data-attribute approach and refactor to rely on `pipe` instead 🙂 

## Related Tickets & Documents

N/A

## QA Instructions, Screenshots, Recordings

Not much to QA here, CI tests should be green

### UI accessibility concerns?

N/A

## Added/updated tests?

- [X] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://developers.forem.com) or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] This PR changes the Forem platform and our documentation needs to be
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
- [ ] I've updated the README or added inline documentation
- [ ] I've added an entry to
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [X] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [ ] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

### CODE REVIEW

File: app/javascript/actionsPanel/initializeActionsPanelToggle.js [hunk 1]
```diff
@@ -29,9 +29,11 @@ export function initializeActionsPanel(user, path) {
     const panelDocument = modContainer.contentDocument;
 
     if (panelDocument) {
-      panelDocument
-        .getElementsByClassName('close-actions-panel')[0]
-        .classList.remove('hidden');
+      const closePanel = panelDocument.getElementsByClassName(
```
Comment (aitchiss) [inline app/javascript/actionsPanel/initializeActionsPanelToggle.js:32]: Locally the spec was occasionally failing due to this closePanel being undefined, so I've added an extra guard here too

File: cypress/integration/seededFlows/moderationFlows/adjustPostTags.spec.js [hunk 2]
```diff
@@ -100,6 +100,9 @@ describe('Adjust post tags', () => {
   });
 
   describe('from article page', () => {
+    // Helper function for pipe command
+    const click = ($el) => $el.click();
```
Comment (aitchiss) [inline cypress/integration/seededFlows/moderationFlows/adjustPostTags.spec.js:104]: The `pipe` command should be passed a named function to allow it to show up in logs properly

File: cypress/integration/seededFlows/moderationFlows/adjustPostTags.spec.js [hunk 3]
```diff
@@ -117,7 +120,13 @@ describe('Adjust post tags', () => {
 
       cy.findByRole('button', { name: 'Moderation' }).click();
       cy.getIframeBody('#mod-container').within(() => {
-        cy.findByRole('button', { name: 'Open adjust tags section' }).click();
+        // Click listeners are attached async so we use pipe() to retry click until condition met
+        cy.findByRole('button', {
+          name: 'Open adjust tags section',
+        })
+          .pipe(click)
+          .should('have.attr', 'aria-expanded', 'true');
```
Comment (aitchiss) [inline cypress/integration/seededFlows/moderationFlows/adjustPostTags.spec.js:128]: The click event will be retried until the button's state updates to 'expanded'. It's quite nice that this pattern allows us to lean on testing good semantics/accessibility 😄  

(NB: the click event will only continue retrying according to the default timeout for the `should` condition)

**Review by nickytonline — APPROVED** [review]
I read the blog post and this all makes sense to me. The only thing I would add to the docs about this is that it's really for a situation like this flaky test. The norm should be to not pipe.

**Comment by aitchiss** [comment]
> I read the blog post and this all makes sense to me. The only thing I would add to the docs about this is that it's really for a situation like this flaky test. The norm should be to not pipe.

Yeah 100%. I think specifically this should be used for those edge cases where click events are added async, but definitely should be an exception rather than a rule. I can see it being useful for some of the dropdown tests too. I kinda like how `pipe` isn't an obvious function name for trying to click something, so hopefully it should be something people arrive at after stumbling across flake.

I'll tag you in the PR when I update the docs so we can get some good guidance in around this.

**Review by djuber — APPROVED** [review]
